specification/integration_context.md

```diff
@@ -0,0 +1,86 @@
+<!-- markdownlint-configure-file 
+{ 
+  "MD013": { 
+    "tables": false 
+  } 
+}
+-->
+
+# Integration Context
+
+**Status**: [Experimental](../README.md#versioning-and-status-of-the-specification)
+
+In order to create a more complete picture of a distributed environment, we
+propose the use of additional headers to pass extra context details between
+Cisco observability components. Specifically, we are concerned with passing
+information between OpenTelemetry-based Splunk distributions and AppDynamics
+agents, in both directions.
+
+# Headers
+
+The following defines several new headers and describes their intended use.
+All headers MUST be treated as optional -- peer services will not always
+generate them.
+
+* `cisco-ctx-acct-id` - Contains the ID of the AppDynamics account.
+* `cisco-ctx-app-id` - Contains the ID of the AppDynamics application.
+* `cisco-ctx-tier-id` - Contains the ID of the AppDynamics tier.
+* `cisco-ctx-bt-id` - Contains the ID of the AppDynamics business transaction (BT).
+* `cisco-ctx-env` - Contains
+  the [deployment.environment.name](https://opentelemetry.io/docs/specs/semconv/attributes-registry/deployment/)
+  resource value from an OpenTelemetry based component.
+* `cisco-ctx-service` - Contains the [service.name](https://opentelemetry.io/docs/specs/semconv/resource/#service)
+  resource value from an OpenTelemetry based component.
+
+HTTP headers are capable of being multivalued. As such, implementations
+SHOULD use the _last_ value when the above headers contain multiple values.
+
+## Splunk OpenTelemetry distributions
+
+This section applies to Splunk distributions of OpenTelemetry instrumentation
+components. In brief, Splunk OTel distributions will need to understand when
+spans are being created as part of an AppDynamics "Business Transaction" (BT),
+and will add span attributes to clarify the AppDynamics invocation context.
+
+### Configuration
+
+Splunk OTel distributions MUST provide a way for users to opt-into the
+consumption and propagation of additional Cisco (bespoke) integration context.
+
+The configuration SHOULD be named `cisco.ctx.enabled` or an idiomatic
+equivalent for each language. If using environment variables, the
+configuration SHOULD be named `CISCO_CTX_ENABLED`. The default value
+MUST be `false` or equivalent language-specific non-truthy value.
+
+## Incoming State
+
+When `cisco.ctx.enabled` is `true`, Splunk implementations MUST
+extract fields from the `cisco-ctx-*` headers (above) and add extra
+attributes to any Spans created as part of the incoming request context.
+Null or missing values MUST be handled gracefully by simply
+omitting the span attributes.
```

Comment:
Before merging, we should create POC for this.
I do no like the idea of adding new attributes to every span. Why not just to the first span in the application? Other calculation could be done by the standard trace-id, span-id and the hierarchy.

Reply:
I think it needs to be on every span for that request. It is more concise that way. 

You also have to allow for a scenario where a call enters the app, leaves the app, and comes back in from a different tier. Multiple tiers possibly for the same trace from different tiers.

You said that you don't like adding attributes, but you didn't say why.